### PR TITLE
Task/elliottyates/tlt 1314 1979/jasmine tests

### PR DESCRIPTION
**Pull request by elliottyates**

### CODE REVIEW

File: course_info/static/course_info/js/controllers/PeopleController.js [hunk 1]
```diff
@@ -24,6 +24,7 @@
              to the course, or null if the add was not attempted for one of the
              reasons listed above.
              */
+            //console.log('----> '+personResult[0]);
```
Comment (elliottyates) [inline course_info/static/course_info/js/controllers/PeopleController.js]: axe

File: course_info/tests/jasmine/controllers/PeopleControllerSpec.js [hunk 2]
```diff
@@ -200,13 +200,26 @@ describe('Unit testing PeopleController', function() {
...
+                    email_address: "test45@mcelroy.org",
+                    name_first: "Test",
+                    name_last: "User",
+                    prime_role_indicator: "Y",
+                    role_type_cd: "STUDENT",
+                    univ_id: "12345678"
+                }
+                ["test45@mcelroy.org"]
+            ];
             spyOn(scope, 'filterSearchResults').and.returnValue([personResult[0]]);
```
Comment (elliottyates) [inline course_info/tests/jasmine/controllers/PeopleControllerSpec.js]: I think these larger changes to this file can be avoided if we just change this one line to 
```
spyOn(scope, 'filterSearchResults').and.returnValue(personResult);
```

File: cross_list_courses/tests/jasmine/controllers/ListControllerSpec.js [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+        $compile, djangoUrl, $log, $q, $uibModal;
+
+    var controller, scope;
+    var xlistURL =
+        '/angular/reverse/?djng_url_name=icommons_rest_api_proxy&djng_url_args' +
+        '=api%2Fcourse%2Fv2%2Fxlist_maps%2F';
+
+
+    function clearInitialxlistFetch() {
+        // handle the initial course instance get
```
Comment (elliottyates) [inline cross_list_courses/tests/jasmine/controllers/ListControllerSpec.js]: comment needs to be updated

File: cross_list_courses/tests/jasmine/controllers/ListControllerSpec.js [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+        beforeEach(function(){
+            spyOn(scope, 'clearMessages');
+            spyOn(scope, 'removeCrosslisting');
+            scope.confirmRemove(xlistMap);
+            $httpBackend.expectGET("partials/remove-xlist-map-confirmation.html").respond(200, '');
+            $httpBackend.flush(1);
+            scope.$digest();
+        });
+        it('should show the modal dialog with correct data when the user clicks delete', function(){
+            for (modalScope = scope.$$nextSibling; modalScope != null; modalScope = modalScope.$$nextSibling) {
```
Comment (elliottyates) [inline cross_list_courses/tests/jasmine/controllers/ListControllerSpec.js]: we could add the initial `var` declaration for `modalScope`

File: cross_list_courses/tests/jasmine/controllers/ListControllerSpec.js [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+            };
+        beforeEach(function () {
+            scope.rawFormInput.primary = primary;
+            scope.rawFormInput.secondary = secondary;
+            scope.dtInstance = {reloadData: function(){}};
+            scope.submitAddCrosslisting();
+        });
+
+        it('should make sure the postNewCrosslisting is called with the correct values', function(){
+            $httpBackend.expectPOST(xlistURL, expectedPostParams)
```
Comment (elliottyates) [inline cross_list_courses/tests/jasmine/controllers/ListControllerSpec.js]: We're actually testing postNewCrosslisting here. To isolate this test, we would want to mock it like we do for removeCrosslisting().

File: cross_list_courses/tests/jasmine/controllers/ListControllerSpec.js [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+        it('shows correct scope.message on failure', function(){
+            postNewCrosslistingDeferred.reject('failure!');
+            scope.$digest();
+
+            expect(scope.operationInProgress).toBeNull();
+            expect(scope.dtInstance.reloadData).not.toHaveBeenCalled();
+            expect(scope.handleAjaxErrorResponse).toHaveBeenCalled();
+            expect(scope.message.alertType).toBe('danger');
+        });
+
```
Comment (sapnamysore) [inline cross_list_courses/tests/jasmine/controllers/ListControllerSpec.js]: Would it make sense to add a  test case for the special error message handling for the duplicate? (most messages come from backend, but the one that is handled in postFailed for the duplicate)
https://github.com/Harvard-University-iCommons/canvas_account_admin_tools/blob/develop/cross_list_courses/static/cross_list_courses/js/controllers/ListController.js#L140

**Comment by sapnamysore** [comment]
Other than the question about an additional test case, looks good 👍

File: cross_list_courses/tests/jasmine/controllers/ListControllerSpec.js (hunk 6 above)
Comment (elliottyates) [inline cross_list_courses/tests/jasmine/controllers/ListControllerSpec.js]: yes! good thought.
